### PR TITLE
Handling FileNotFoundError exception on sigterm

### PR DESCRIPTION
**Pull request by vyscond**
For some reason after the sigterm signal being handled the `Daemonize.exit` function is called raising the FileNotFoundError over the the trying of removing the pid file (already removed over sigterm).

Since it was already removed i think it's just a simple straight forward solution just to warn up the developer? Well. this is relate to issue #43

### CODE REVIEW

**Comment by thesharp** [comment]
Hello @vyscond 

We can't use print there because we already closed our FDs. I think we should just stop this situation from happening. We can register something like `self.pidfile_removed = True` after pid file removal inside the `sigterm()`. By checking aforementioned value we can remove our pid file inside the `exit()` only if `self.pidfile_removed` value equals to False.

**Comment by vyscond** [comment]
1 - FDs? 

2 - If Daemonize need to call `sigterm` and than `exit` i think only one could handle it so. Afaik no need to flag it.  I think you can remove the `print` and add a `pass`? Or even a custom handler/callback in case the pid was deleted by other process by mistake... Exceptions are here to help us! No need to go back in C and create flags.

3 - Why Daemonize call this two function in a row? even more suspicious why calling two times `sys.exit`?

File: daemonize.py [hunk 1]
```diff
@@ -68,7 +68,10 @@ def exit(self):
         Cleanup pid file at exit.
         """
         self.logger.warn("Stopping daemon.")
-        os.remove(self.pid)
+        try:
+            os.remove(filename)
+        except FileNotFoundError:
```
Comment (knsd) [inline daemonize.py]: `FileNotFoundError` exception was introduced in Python 3.3, see PEP 3151, so it's broken almost always.

**Comment by thesharp** [comment]
@vyscond,

> 1

https://en.wikipedia.org/wiki/File_descriptor

> If Daemonize need to call sigterm and than exit then i think only one could/need to handle the 
> os.remove function.

We register both `exit()` and `sigterm()` near the end of `daemonize.py` to be able to correctly handle both situation which are completely different. Former handles normal shutdown, whilst latter handles forced one, triggered by the user. Arguably, we can try to remove the `os.remove()` bit from `sigterm()` entirely, because at the end of it we call `sys.exit(0)`, which triggers `exit()` function with another `os.remove()` in place.

> 3

I think I've addressed that issue in the above paragraph.

@knsd, what do you think? Should we entirely drop pid file removal from `sigterm()`, because it will happen inside `exit()` no matter if `sigterm()` was triggered or not.

**Comment by knsd** [comment]
> @knsd, what do you think? Should we entirely drop pid file removal from sigterm(), because it will happen inside exit() no matter if sigterm() was triggered or not.

👍

**Comment by thesharp** [comment]
Okay, let's do it.

**Comment by vyscond** [comment]
at least was solved :)

**Pull request closed without merging**